### PR TITLE
Creating a simple Java project as a lead receiver sample

### PR DESCRIPTION
**Pull request by giovannicaldeira**
## What changes were proposed in this pull request?

Creating a simple Java project as a lead receiver sample

## How was this patch tested?

This sample was created to test this feature: https://developers.grupozap.com/leads/endpoint-validator/

Issue: https://github.com/olxbr/squad-listing-integration/issues/1645

### CODE REVIEW

File: samples/java/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+### Cloning
+
+``` shell
+git clone https://github.com/olxbr/crm-lead-integration.git
+cd samples/java
+```
+
+### Setting SECRET_KEY
+
+Change `SECRET_KEY` valeu in [application.properties](src/main/resources/application.properties) file
```
Comment (marcieltorres) [inline samples/java/README.md:25]:
```suggestion
Change `SECRET_KEY` value in [application.properties](src/main/resources/application.properties) file
```

File: samples/java/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Lead Integration - Java
```
Comment (marcieltorres) [inline samples/java/README.md:1]: How about creating a ptbr version of readme? 


`https://github.com/olxbr/crm-lead-integration/blob/master/samples/nodejs/README-ptbr.md`

File: samples/java/src/main/java/br/com/example/leadIntegration/controllers/LeadIntegrationController.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+@RequestMapping("/leads")
+public class LeadIntegrationController {
+
+    @GetMapping("/health-check")
+    public String healthCheck() {
+        return "{\"message\": \"OK\"}";
+    }
+
+    @PostMapping("/lead")
+    public String lead(@RequestBody Lead lead){
```
Comment (marcieltorres) [inline samples/java/src/main/java/br/com/example/leadIntegration/controllers/LeadIntegrationController.java:16]: Do you are checking the `SECRET_KEY? :thinking: 

something like this: `https://github.com/olxbr/crm-lead-integration/blob/master/samples/python/app/api.py#L19`

File: samples/java/README.md (hunk 2 above)
Comment (giovannicaldeira) [reply samples/java/README.md:1]: Added ptbr version:

`samples/java/README-ptbr.md`

File: samples/java/src/main/java/br/com/example/leadIntegration/models/Lead.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
```
Comment (robsonoduarte) [inline samples/java/src/main/java/br/com/example/leadIntegration/models/Lead.java:11]: @giovannicaldeira maybe you can change all this annotation for this https://projectlombok.org/features/Data

File: samples/java/src/main/java/br/com/example/leadIntegration/controllers/LeadIntegrationController.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package br.com.example.leadIntegration.controllers;
+
+import org.springframework.web.bind.annotation.*;
+import br.com.example.leadIntegration.models.Lead;
+
+@RestController
+@RequestMapping("/leads")
+public class LeadIntegrationController {
+
+    @GetMapping("/health-check")
```
Comment (robsonoduarte) [inline samples/java/src/main/java/br/com/example/leadIntegration/controllers/LeadIntegrationController.java:10]: @giovannicaldeira use the `actuator` to create the default springboot health check https://www.baeldung.com/spring-boot-actuators

File: samples/java/src/test/java/br/com/example/leadIntegration/LeadIntegrationApplicationTests.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+	}
+
+	@Test
+	public void recieveLeadShouldBeSuccess() throws Exception {
+
+		final String baseUrl = "http://localhost:"+port+"/leads/lead";
+		URI uri = new URI(baseUrl);
+
+		Lead lead = new Lead("VivaReal", "2017-10-23T15:50:30.619Z", "59ee0fc6e4b043e1b2a6d863", "87027856",
```
Comment (robsonoduarte) [inline samples/java/src/test/java/br/com/example/leadIntegration/LeadIntegrationApplicationTests.java:53]: @giovannicaldeira for more fluent code ( in my opinion ) we can use the builder pattern, the lombok do this magic for us https://projectlombok.org/features/Builder

File: samples/java/README-ptbr.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+Um simples projeto usando Java para demonstrar como receber leads do zap+.
+
+Por favor, confira nossa [documentação](https://developers.grupozap.com/) para mais informações.
+
+## Recursos Utilizados
+
+- [Java 11](https://openjdk.java.net/projects/jdk/11/)
+- [Maven 3.8.1](https://maven.apache.org/)
+- [Spring Boot 2.5.0](https://spring.io/projects/spring-boot)
+- [Docker](https://www.docker.com/get-started)
```
Comment (robsonoduarte) [inline samples/java/README-ptbr.md:14]: @giovannicaldeira put the lombok in this list

File: samples/java/pom.xml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>2.5.0</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>br.com.example</groupId>
```
Comment (robsonoduarte) [inline samples/java/pom.xml:11]: @giovannicaldeira we can use the olx.com.br in the groupId

File: samples/java/pom.xml [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+	<artifactId>leadIntegration</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
```
Comment (robsonoduarte) [inline samples/java/pom.xml:13]: @giovannicaldeira use the 0.0.1 for version

File: samples/java/pom.xml [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>2.5.0</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>br.com.example</groupId>
+	<artifactId>leadIntegration</artifactId>
```
Comment (robsonoduarte) [inline samples/java/pom.xml:12]: @giovannicaldeira for projects in java we can use the names with `-` like this https://github.com/olxbr/listings-api

File: samples/java/src/main/java/br/com/example/leadIntegration/models/Lead.java (hunk 4 above)
Comment (giovannicaldeira) [reply samples/java/src/main/java/br/com/example/leadIntegration/models/Lead.java]: @robsonoduarte Change annotation to Data 

```java
@Data
@Builder
public class Lead {
...
}
```

File: samples/java/src/main/java/br/com/example/leadIntegration/controllers/LeadIntegrationController.java (hunk 5 above)
Comment (giovannicaldeira) [reply samples/java/src/main/java/br/com/example/leadIntegration/controllers/LeadIntegrationController.java]: @robsonoduarte added `actuator` to project. Remove `health-check` endpoint.

File: samples/java/src/test/java/br/com/example/leadIntegration/LeadIntegrationApplicationTests.java (hunk 6 above)
Comment (giovannicaldeira) [reply samples/java/src/test/java/br/com/example/leadIntegration/LeadIntegrationApplicationTests.java]: @robsonoduarte added Builder annotation.

```java
Lead lead = Lead.builder()
				.leadOrigin("VivaReal")
				.timestamp("2017-10-23T15:50:30.619Z")
				.originLeadId("59ee0fc6e4b043e1b2a6d863")
				.originListingId("87027856")
				.clientListingId("a40171")
				.name("Nome Consumidor")
				.email("nome.consumidor@email.com")
				.ddd("11")
				.phone("999999999")
				.message("Olá, tenho interesse neste imóvel. Aguardo o contato. Obrigado.")
				.build();
```

File: samples/java/README-ptbr.md (hunk 7 above)
Comment (giovannicaldeira) [reply samples/java/README-ptbr.md:15]: @robsonoduarte added lombok.

File: samples/java/README.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+A basic sample using Java to show how to receive leads from zap+.
+
+Please, check out our [docs](https://developers.grupozap.com/) for more information.
+
+## Technology and Resources
+
+- [Java 11](https://openjdk.java.net/projects/jdk/11/)
+- [Maven 3.8.1](https://maven.apache.org/)
+- [Spring Boot 2.5.0](https://spring.io/projects/spring-boot)
+- [Docker](https://www.docker.com/get-started)
```
[the remark this answers is not in the document]
Comment (giovannicaldeira) [reply samples/java/README.md:15]: @robsonoduarte added lombok.

File: samples/java/pom.xml (hunk 8 above)
Comment (giovannicaldeira) [reply samples/java/pom.xml]: @robsonoduarte changed  groupId

File: samples/java/pom.xml (hunk 9 above)
Comment (giovannicaldeira) [reply samples/java/pom.xml]: @robsonoduarte changed version to `0.0.1`

File: samples/java/pom.xml (hunk 10 above)
Comment (giovannicaldeira) [reply samples/java/pom.xml]: @robsonoduarte changed to `lead-integration`

**Comment by robsonoduarte** [comment]
@giovannicaldeira what do you think to put gradle's build on the readme, makefile.. ? :eyes:

**Comment by giovannicaldeira** [comment]
@robsonoduarte gradle reference created.

**Review by robsonoduarte — APPROVED** [review]

**Review by eduardomatoss — APPROVED** [review]

**Pull request merged**